aip/general/0193.md

```diff
@@ -75,132 +86,191 @@ following:
-`ErrorInfo` payload in the `details` field. Variable information
-**should** be included in the `metadata` field on `ErrorInfo` and
-**must** be included if it appears within an error message.
+**Note:** Sometimes a service will elect to always present
+`Status.message` in English rather than the application's native
+language so that messages are easily searchable in common knowledge
+bases, such as StackOverflow™.
 
-#### Uniqueness
+Important points:
 
-Each type of detail payload **must** be included at most once. For
-example, there **must not** be more than one [`BadRequest`][BadRequest]
-message in the `details` field, but there **may** be a `BadRequest` and
-a [`PreconditionFailure`][PreconditionFailure].
+- This `message` string is considered a problem description. It is
+  intended for human developers to understand the problem and is more
+  detailed than [`ErrorInfo.reason`][ErrorInfo-reason], discussed in the
+  next section.
+- Use simple descriptive language that is easy to understand (without
+  technical jargon) to clearly state the problem that results in an
+  error.
+- For pre-existing (brownfield) APIs, the information in
+ `google.rpc.Status` already exists and can’t be changed. In this
+ case, do not edit anything in this `google.rpc.Status`, because some
+ users might have automation running against the pre-existing
+ content.  For example, any changes in text for the `message` string
+ could cause breakages. For more information, see [Changing Error
+ Messages](#changing-error-messages).
 
-**Note:** `ErrorInfo` represents a special case. There **must** be exactly one
-`ErrorInfo`. It is required.
 
-### Error messages
+##### [google.rpc.Status payload][Status]
 
-For each error, the service **must** populate the `message` field on
-[`google.rpc.Status`][Status]. This error message,
+Details of this object are summarized in the following fields, field
+descriptions, and examples:
 
-- is a developer-facing, human-readable "debug message"
-- is presented in the service's native language
-- both explains the error and offers an actionable resolution to it
-  ([citation](https://cloud.google.com/apis/design/errors#error_model))
 
-**Note:** Sometimes a service will elect to always present
-`Status.message` in English rather than the application's native
-language so that messages are easily searchable in common knowledge
-bases, such as StackOverflow&trade;.
+- `code int32`
+  - The status code, which **should** be an enum value of
+    [`google.rpc.Code`][Code].
+  - *Example*: 429
+- `message string`
+  - A developer-facing error message, which **should** be in English.
+    Any user-facing error message **should** be located in the
+    LocalizedMessage payload.
+  - *Example message*: \"The zone \'us-east1-a\' does not have enough
+    resources available to fulfill the request. Try a different zone, or
+    try again later.\"
+- `details Object (repeated)`
+  - Text
```

Comment:
I think "Text" here is confusing. What does it mean in this context? `details` is a repeated Object in JSON and a repeated Message in proto. I don't see what value the word, "Text", adds here. It is an overloaded term. e.g., some databases have a text column type.

Reply:
Can we just remove the line with - Text?

Reply:
I think it's still beneficial to have a description here to be consistent with the other bullet points. How about something along the lines of "Additional details about the error, which include ErrorInfo, LocalizedMessage, etc."?

Reply:
Updated.